app/controllers/agent.js

```diff
@@ -28,4 +28,13 @@ export default Ember.Controller.extend(DetailMixin, {
     outgoingAffiliations: Ember.computed('model.outgoingAgentRelations', function() {
         return this.get('model.outgoingAgentRelations').map(relation => relation.related);
     }),
+
+    actions: {
+        goBack() {
+            if (document.referrer.includes('\/share')) {
+                history.back();
+            }
+            this.transitionToRoute('discover');
```

Comment:
Should this be wrapped in an `else`? It's my understanding that `history.back()` is more or less equivalent to pressing back. I could see some odd behavior coming up if both happen to be hit.